src/JsonSchema/Constraints/BaseConstraint.php

```diff
@@ -153,4 +153,16 @@ public static function arrayToObjectRecursive($array)
 
         return (object) json_decode($json);
     }
+
+    /**
+     * Transform a JSON pattern into a PCRE regex
+     *
+     * @param string $pattern
+     *
+     * @return string
+     */
+    public static function jsonPatternToPhpRegex($pattern)
+    {
+        return '#' . str_replace('#', '\\#', $pattern) . '#u';
```

Comment:
While at it, I propose to use a safer delimiter (`#` is considered special by `preg_quote()` since PHP 7.3), say `~` (not `/` because it needs to be escaped currently). WDYT?